consensus/state.go

```diff
@@ -48,6 +48,12 @@ type msgInfo struct {
 	PeerID p2p.ID  `json:"peer_key"`
 }
 
+// indicate to the reactor that a peer should be removed
+type badPeerInfo struct {
+	Error  error  `json:"error"`
+	PeerID p2p.ID `json:"peer_key"`
+}
```

Comment:
`error` [doesn't marshal effectively](https://go.dev/play/p/xr81vCBqhTM).

```go
package main

import (
	"encoding/json"
	"errors"
	"fmt"
)

type badPeerInfo struct {
	Error error `json:"error"`
}

func main() {
	i := badPeerInfo{
		Error: errors.New("this is an error"),
	}

	data, err := json.Marshal(i)
	if err != nil {
		panic(err)
	}

	fmt.Printf("%s\n", string(data))
}
```

```
{"error":{}}
```

Reply:
Wow thanks. What is the right way to do this ? Just use a string in the struct?

Reply:
> Wow thanks. What is the right way to do this ? Just use a string in the struct?

[Yeah, basically](https://go.dev/play/p/Py-SPvqajEH), e.g.

```go
type badPeerInfo struct {
	Error string `json:"error,omitempty"` // omitempty is important I guess
}
```